lib/gemstash/gem_source/upstream_source.rb

```diff
@@ -160,7 +160,15 @@ def fetch_local_gem(gem_name, gem_resource, resource_type)
       def fetch_remote_gem(gem_name, gem_resource, resource_type)
         log.info "Gem #{gem_name.name} is not cached, fetching #{resource_type}"
         gem_fetcher.fetch(gem_name.id, resource_type) do |content, properties|
-          gem_resource.save({ resource_type => content }, headers: { resource_type => properties })
+          resource_properties = {
+            upstream: upstream.to_s,
+            gem_name: gem_name.name,
+            headers: { resource_type => properties }
+          }
+
+          gem = gem_resource.save({ resource_type => content }, resource_properties)
+          Gemstash::DB::CachedRubygem.store(upstream, gem_name, resource_type)
```

Comment:
That's typically how I deal with these sorts of methods. I like it that way because I can hide any transactions or nested creations or lookups in the class method, along with extracting the proper information out of the various objects passed in. I think of it like a factory method basically, which seems appropriate as a class method. But to be honest, I'm not sure what would happen if we tried to write a custom initialize method for something that extends `Sequel::Model`...